greeter/README.md

```diff
@@ -1,21 +1,706 @@
+2. Restart the application again by running `npm start`.
+3. Go to API Explorer, http://localhost:3000/explorer.
+4. Under the `GET /greet/{name}` endpoint, enter `fr` as the language as shown below.
+
+   ![Test French Greeter](images/french-greeter-test.png)
+
+5. In the response body section, you should be able to see the greeting in French.
+   ```json
+   {
+     "timestamp": "2019-10-01T18:47:37.096Z",
+     "language": "fr",
+     "greeting": "Diana, Bonjour"
+   }
+   ```
+
+### Step 4: Configure the Greeter
+
+Besides from compositing your app with modules and extensions, LB4 also allows you to configure bound items in the context easily. Such configurations can be resolved and injected in the same way of other forms of dependencies.
+
+We will show how to configure the way it prints the Chinese greeting message with name first, e.g. `"LoopBack, hello"`. (The default setting is greeting word `"Hello"` first, e.g. `"hello, LoopBack"`)
+
+Add this line in your `application.ts` file
+
+```ts
+this.component(GreetingComponent);
+this.add(createBindingFromClass(FrenchGreeter));
+
+// Add the following line
+this.configure('greeters.ChineseGreeter').to({nameFirst: false});
+```
+
+### Step 5: Try it Out!
+
+Restart your application, try endpoint `GET /greeting/{name}`, with `Accept-Language` set to `zh`.
+
+You will see message `'name, hello'` printed instead of `'hello, name'`.
+
+Related documentation:
+[Configuration by convention](https://loopback.io/doc/en/lb4/Context.html#configuration-by-convention)
+
+## Part 3: Enable Caching
+
+To reduce the cost of calculation, we'd like to use a caching server for our greeting application. We will show how to bind a caching server, interceptor and observer for your application as an example of contributing microservice using LB4.
+
+The caching server will cache the greeting messages for a certain amount of time. And cached message will be swept out when they are expired.
+
+Again, we are using `bind` and `injection` to bind our app with Interceptor and Observer:
+
+- Interceptors: are reusable functions to provide aspect-oriented logic around method invocations. (should make it more intuitive)
+- Observer: is a LoopBack application that has its own life cycles at runtime. Observer allows artifacts to participate in the life cycles. The transition of states is triggered by two functions: start() and stop().
+
+### Step 1: Create a caching service
+
+1.  Assuming this caching service is ready to use, we'll just copy and paste `src/services/caching.service.ts`.
+
+2.  Create the binding key for caching service.
+
+As we mentioned above, a binding connects its value to a unique key as the address to access the entry in a context. For example,
+
+```ts
+ctx.bind('hello').to('world'); // BindingKey='hello', BindingValue='world'`
+```
+
+Let's create the binding key for caching service so that you can inject it in an observer or an interceptor later by the binding key.
```

Comment:
> Let's create the binding key for caching service so that you can inject it in an observer or an interceptor later by the binding key.

Let's create the binding key for the caching service so that you can inject it in an observer or an interceptor later.